test/loopback-dl.test.js

```diff
@@ -837,6 +837,81 @@ describe('Models attached to a dataSource', function() {
 
   });
 
+  context('replaceAttributes', function() {
+    it('should replace with new data', function(done) {
+      Post.create({title: 'a', content: 'AAA', comments: ['Comment1']},
+      function(err, post) {
+        if (err) return done(err);
+        post.replaceAttributes({title: 'b'}, function(err, p) {
+          if (err) return done(err);
+          p.id.should.equal(post.id);
+          should.not.exist(p._id);
```

Comment:
Please use `p.should.not.have.property('_id')`, as it produces much more helpful error messages on failure. Apply this change to all new test code please.